core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -757,7 +757,8 @@ private UnifyResult matchRecurse(MutableRel target) {
 
   private UnifyResult apply(UnifyRule rule, MutableRel query,
       MutableRel target) {
-    final UnifyRuleCall call = new UnifyRuleCall(rule, query, target, null);
+    final UnifyRuleCall call =
```

Comment:
This code path is never called. However, `UnifyRuleCall` requires nonNull `slots`.
```
    public UnifyRuleCall(UnifyRule rule, MutableRel query, MutableRel target,
        ImmutableList<MutableRel> slots) {
      this.rule = Objects.requireNonNull(rule);
      this.query = Objects.requireNonNull(query);
      this.target = Objects.requireNonNull(target);
      this.slots = Objects.requireNonNull(slots);
    }
```